pinot-connectors/pinot-flink-connector/src/main/java/org/apache/pinot/connector/flink/sink/FlinkSegmentWriter.java

```diff
@@ -95,8 +100,12 @@ public class FlinkSegmentWriter implements SegmentWriter {
   private transient Counter _processedRecords;
   private transient volatile long _lastRecordProcessingTimeMs = 0;
 
-  public FlinkSegmentWriter(int indexOfSubtask, MetricGroup metricGroup) {
+  public FlinkSegmentWriter(int indexOfSubtask, MetricGroup metricGroup, String segmentNamePrefix,
```

Comment:
b/w incompatible change is not a concern here? I guess callers only call the sink function?

Reply:
right, PinotSinkFunction is exposed for Flink and not a whole lot of benefit in using FlinkSegmentWriter directly.
There is a `FileBasedSegmentWriter` implementing the interface `SegmentWriter` for general use.

Still, let me call this out in the description and mark backward incompatible.